### PR TITLE
Add a license

### PR DESCRIPTION
**Pull request by mrigger**
Addresses #16.

### CODE REVIEW

**Comment by lukasstadler** [comment]
looks good - ideally you could also include the license header in all files to which this license applies, to make it clear.
either in full, or a short version as shown in https://oss.oracle.com/licenses/upl/#_How_should_I

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+This license is subject to the following condition:
+
+The above copyright notice and either this complete permission notice or at a minimum a reference to the UPL must be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+The Benchmarks
+==============
+The benchmarks in the `benchmark` subdirectory are not distributed under the UPL license and have their own licenses.
```
Comment (pointhi) [inline LICENSE.md]: The Benchmark Harness itself also has a different license

Comment (mrigger) [reply LICENSE.md]: Thanks! I updated the `LICENSE.md` accordingly.

**Comment by mrigger** [comment]
> looks good - ideally you could also include the license header in all files to which this license applies, to make it clear.
> either in full, or a short version as shown in https://oss.oracle.com/licenses/upl/#_How_should_I

Thanks for the feedback @lukasstadler ! I intend to do so, but I also want to have a check in place so that this is enforced. For this PR, I first want to include the `LICENSE.md`.

**Pull request merged**